lighthouse-core/config/config-helpers.js

```diff
@@ -83,35 +160,206 @@ function assertValidAudit(auditDefinition) {
   }
 }
 
+/**
+ * Expands a gatherer from user-specified to an internal gatherer definition format.
+ *
+ * Input Examples:
+ *  - 'my-gatherer'
+ *  - class MyGatherer extends Gatherer { }
+ *  - {instance: myGathererInstance}
+ *
+ * @param {LH.Config.GathererJson} gatherer
+ * @return {{instance?: Gatherer, implementation?: GathererConstructor, path?: string}} passes
+ */
+function expandGathererShorthand(gatherer) {
+  if (typeof gatherer === 'string') {
+    // just 'path/to/gatherer'
+    return {path: gatherer};
+  } else if ('implementation' in gatherer || 'instance' in gatherer) {
+    // {implementation: GathererConstructor, ...} or {instance: GathererInstance, ...}
+    return gatherer;
+  } else if ('path' in gatherer) {
+    // {path: 'path/to/gatherer', ...}
+    if (typeof gatherer.path !== 'string') {
+      throw new Error('Invalid Gatherer type ' + JSON.stringify(gatherer));
+    }
+    return gatherer;
+  } else if (typeof gatherer === 'function') {
+    // just GathererConstructor
+    return {implementation: gatherer};
+  } else if (gatherer && typeof gatherer.beforePass === 'function') {
+    // just GathererInstance
+    return {instance: gatherer};
+  } else {
+    throw new Error('Invalid Gatherer type ' + JSON.stringify(gatherer));
+  }
+}
+
 /**
  * Expands the audits from user-specified JSON to an internal audit definition format.
- * @param {LH.Config.Json['audits']} audits
- * @return {?Array<{id?: string, path: string, options?: {}} | {id?: string, implementation: typeof Audit, path?: string, options?: {}}>}
+ * @param {LH.Config.AuditJson} audit
+ * @return {{id?: string, path: string, options?: {}} | {id?: string, implementation: typeof Audit, path?: string, options?: {}}}
  */
-function expandAuditShorthand(audits) {
-  if (!audits) {
-    return null;
+function expandAuditShorthand(audit) {
```

Comment:
changed to operate on an individual audit instead of an array for consistency with gatherers, otherwise untouched